src/main/java/org/cryptomator/linux/quickaccess/DolphinPlaces.java

```diff
@@ -0,0 +1,172 @@
+		} catch (IOException | SAXException e) {
+			throw new IllegalStateException("Failed to load included XBEL schema definition file.", e);
+		}
+	}
+
+
+	@Override
+	public QuickAccessService.QuickAccessEntry add(Path target, String displayName) throws QuickAccessServiceException {
+		String id = UUID.randomUUID().toString();
+		try {
+			MODIFY_LOCK.lock();
+			if (Files.size(PLACES_FILE) > MAX_FILE_SIZE) {
+				throw new IOException("File %s exceeds size of %d bytes".formatted(PLACES_FILE, MAX_FILE_SIZE));
+			}
+			var placesContent = Files.readString(PLACES_FILE);
+			//validate
+			xmlValidator.validate(new StreamSource(new StringReader(placesContent)));
+			// modify
+			int insertIndex = placesContent.lastIndexOf("</xbel"); //cannot be -1 due to validation; we do not match the end tag, since betweent tag name and closing bracket can be whitespaces
+			try (var writer = Files.newBufferedWriter(TMP_FILE, StandardCharsets.UTF_8, StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+				writer.write(placesContent, 0, insertIndex);
+				writer.newLine();
+				writer.write(ENTRY_TEMPLATE.formatted(target.toUri(), displayName, id).indent(1));
+				writer.newLine();
+				writer.write(placesContent, insertIndex, placesContent.length() - insertIndex);
+			}
+			// save
+			Files.move(TMP_FILE, PLACES_FILE, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
+			return new DolphinPlacesEntry(id);
+		} catch (SAXException | IOException e) {
+			throw new QuickAccessServiceException("Adding entry to KDE places file failed.", e);
+		} finally {
+			MODIFY_LOCK.unlock();
+		}
+	}
+
+	private static class DolphinPlacesEntry implements QuickAccessEntry {
+
+		private final String id;
+		private volatile boolean isRemoved = false;
+
+		DolphinPlacesEntry(String id) {
+			this.id = id;
+		}
+
+		@Override
+		public void remove() throws QuickAccessServiceException {
+			try {
+				MODIFY_LOCK.lock();
+				if (isRemoved) {
+					return;
+				}
+				if (Files.size(PLACES_FILE) > MAX_FILE_SIZE) {
+					throw new IOException("File %s exceeds size of %d bytes".formatted(PLACES_FILE, MAX_FILE_SIZE));
+				}
+				var placesContent = Files.readString(PLACES_FILE);
+				int idIndex = placesContent.lastIndexOf(id);
+				if (idIndex == -1) {
+					isRemoved = true;
+					return; //we assume someone has removed our entry
+				}
+				//validate
+				xmlValidator.validate(new StreamSource(new StringReader(placesContent)));
+				//modify
+				int openingTagIndex = indexOfEntryOpeningTag(placesContent, idIndex);
+				var contentToWrite1 = placesContent.substring(0, openingTagIndex).stripTrailing();
+
+				int closingTagEndIndex = placesContent.indexOf('>', placesContent.indexOf("</bookmark", idIndex));
+				var part2Tmp = placesContent.substring(closingTagEndIndex + 1).split("\\v+", 2); //removing leading vertical whitespaces, but no indentation
+				var contentToWrite2 = part2Tmp[part2Tmp.length - 1];
+
```

Comment:
Sadly https://github.com/cryptomator/integrations-linux/pull/85#discussion_r1681045011 is coming back to haunt us.
Again I used  `\\s*` instead of `\\v*` to illustrate:
```
"<bookmark>...</bookmark>".split("\\s+", 2) // { "<bookmark>...</bookmark>" } -- Expected result
" <bookmark>...</bookmark>".split("\\s+", 2) // { "", "<bookmark>...</bookmark>" } -- Expected result
" <bookmark> ... </bookmark>".split("\\s+", 2) // { "", "<bookmark> ... </bookmark>" } -- Expected result
"<bookmark> ... </bookmark>".split("\\s+", 2) // { "<bookmark>", "... </bookmark>" } -- Unexpected result
```

This might happen if a different application adds a bookmark like this
```
<bookmark>
    <!-- Cryptomator -->
</bookmark><bookmark>
    <!-- Different application -->
</bookmark>
```
instead of this
```
<bookmark>
    <!-- Cryptomator -->
</bookmark>
<bookmark>
    <!-- Different application -->
</bookmark>
```

Reply:
I changed the matching expression to only match the [beginning of the input](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/util/regex/Pattern.html#begin_input) in https://github.com/cryptomator/integrations-linux/pull/85/commits/e4061a33720c87cd655fd9846cd1b29688fbf4d9